pandas/_libs/lib.pyx

```diff
@@ -759,7 +759,7 @@ cpdef ndarray[object] ensure_string_array(
             out = arr.astype(str).astype(object)
             out[arr.isna()] = na_value
             return out
-        arr = arr.to_numpy()
+        arr = arr.to_numpy(dtype=object)
```

Comment:
Should this only be done for certain dtypes? (integer?) 

(maybe run the `string.py::Construction` benchmarks to check impact, as those were added for this code path https://github.com/pandas-dev/pandas/pull/37371)